`skywire-cli log` hangs

`skywire-cli log`  can hang indefinitely

The order of the fetching of the surveys and transport logs was randomized here to compensate for it

https://github.com/skycoin/skywire/pull/1541

however the issue is actually still present

the reward system log collection run can be viewed here:

https://fiber.magnetosphere.net/log-collection

The process is limited to half an hour via the `timeout` command, though it takes only about 10 minutes for the command to hang on a download attempt.

here is a panic which occurs from an io timeout error on the uptime tracker query with `skywire-cli log`

![image](https://github.com/skycoin/skywire/assets/36607567/7f4737a4-01cd-470e-a729-b62b40238967)


```
[2023-06-24T04:18:47.424765278-05:00] ERROR [log-collecting]: Error while fetching data from uptime service. Error: Get "http://ut.skywire.skycoin.com/uptimes?v=v2": dial tcp 192.46.229.215:80: i/o timeout
[2023-06-24T04:18:47.42483283-05:00] PANIC [log-collecting]: Unable to get data from uptime tracker. error="Cannot get Uptime data"
panic: (*logrus.Entry) 0xc00032e0e0

goroutine 1 [running, locked to thread]:
github.com/sirupsen/logrus.(*Entry).log(0xc00032e070, 0x0, {0xc0005a0570, 0x27})
	github.com/sirupsen/logrus@v1.8.1/entry.go:259 +0x487
github.com/sirupsen/logrus.(*Entry).Log(0xc00032e070, 0x0, {0xc00017dcf0?, 0x0?, 0x0?})
	github.com/sirupsen/logrus@v1.8.1/entry.go:293 +0x4f
github.com/sirupsen/logrus.(*Entry).Panic(0xf19c9c?, {0xc00017dcf0?, 0xc0001ea170?, 0xc000591360?})
	github.com/sirupsen/logrus@v1.8.1/entry.go:331 +0x29
github.com/skycoin/skywire/cmd/skywire-cli/commands/log.glob..func1(0x1b579e0?, {0xedc437?, 0x2?, 0x2?})
	github.com/skycoin/skywire/cmd/skywire-cli/commands/log/root.go:108 +0x40c
github.com/spf13/cobra.(*Command).execute(0x1b579e0, {0xc000499c00, 0x2, 0x2})
	github.com/spf13/cobra@v1.4.0/command.go:860 +0x663
github.com/spf13/cobra.(*Command).ExecuteC(0x1b54a60)
	github.com/spf13/cobra@v1.4.0/command.go:974 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.4.0/command.go:902
github.com/skycoin/skywire/cmd/skywire-cli/commands.Execute()
	github.com/skycoin/skywire/cmd/skywire-cli/commands/root.go:224 +0x6a
main.main()
	github.com/skycoin/skywire/cmd/skywire-cli/skywire-cli.go:9 +0x17

```

--- a/cmd/skywire-cli/commands/log/root.go
+++ b/cmd/skywire-cli/commands/log/root.go
@@ -72,21 +72,6 @@ var logCmd = &cobra.Command{
 			log.Fatal("use of mutually exclusive flags --log and --survey")
 		}
 
-		// Preparing directories
-		if _, err := os.ReadDir("log_collecting"); err != nil {
-			if err := os.Mkdir("log_collecting", 0750); err != nil {
-				log.Panic("Unable to log_collecting directory")
-			}
-		}
-
-		if err := os.Chdir("log_collecting"); err != nil {
-			log.Panic("Unable to change directory to log_collecting")
-		}
-
-		// Create dmsgget instance
-		dg := dmsgget.New(flag.CommandLine)
-		flag.Parse()
-
 		ctx, cancel := cmdutil.SignalContext(context.Background(), log)
 		defer cancel()
 		go func() {
@@ -94,6 +79,24 @@ var logCmd = &cobra.Command{
 			cancel()
 			os.Exit(1)
 		}()
+
+		// Preparing directories
+		if _, err := os.ReadDir("log_collecting"); err != nil {
+			if err := os.Mkdir("log_collecting", 0750); err != nil {
+				log.Error("Unable to log_collecting directory")
+				return
+			}
+		}
+
+		if err := os.Chdir("log_collecting"); err != nil {
+			log.Error("Unable to change directory to log_collecting")
+			return
+		}
+
+		// Create dmsgget instance
+		dg := dmsgget.New(flag.CommandLine)
+		flag.Parse()
+
 		// Set the uptime tracker to fetch data from
 		endpoint := skyenv.UptimeTrackerAddr + "/uptimes?v=v2"
 		if env == "test" {
@@ -105,7 +108,8 @@ var logCmd = &cobra.Command{
 		//Fetch the uptime data over http
 		uptimes, err := getUptimes(endpoint, log)
 		if err != nil {
-			log.WithError(err).Panic("Unable to get data from uptime tracker.")
+			log.WithError(err).Error("Unable to get data from uptime tracker.")
+			return
 		}
 		//randomize the order of the survey collection - workaround for hanging
 		rand.Shuffle(len(uptimes), func(i, j int) {
@@ -119,7 +123,8 @@ var logCmd = &cobra.Command{
 
 		dmsgC, closeDmsg, err := dg.StartDmsg(ctx, log, pk, sk)
 		if err != nil {
-			log.WithError(err).Panic(err)
+			log.Error(err)
+			return
 		}
 		defer closeDmsg()
 
@@ -156,7 +161,8 @@ var logCmd = &cobra.Command{
 					deleteOnError := false
 					if _, err := os.ReadDir(key); err != nil {
 						if err := os.Mkdir(key, 0750); err != nil {
-							log.Panicf("Unable to create directory for visor %s", key)
+							log.Errorf("Unable to create directory for visor %s", key)
+							return
 						}
 						deleteOnError = true
 					}

